### ISSUE TITLE
Training Loss doesn't go below 1

### ISSUE THREAD

**Issue by qian18long** [hpcaitech/Open-Sora#573]
I use my own data and the stage1 training for v1.2. The loss doesn't go below 1. Do you have any suggestions why this happened?

<img width="401" alt="Screenshot 2024-07-02 at 11 31 13 AM" src="https://github.com/hpcaitech/Open-Sora/assets/45865971/754258ad-7d15-441d-8dd9-897c0d5def8d">

**Comment by tinnerhrhe** [comment]
I have a similar problem. I fine-tune open-sora v1.2 on my own dataset using a learning rate of [1e-4, 2e-5]. The loss curves always fluctuate around 1. After about 70k steps of training, I evaluate the model and, unfortunately, find the quality of the generated videos is not expected.

**Comment by JThh** [comment]
It is well likely due to the dataset. Can you probably follow our stage 1 first: https://github.com/hpcaitech/Open-Sora/blob/main/docs/report_03.md#first-stage, before you use your own dataset to continue training?

**Comment by CIntellifusion** [comment]
Maybe you can also check whether the pretrained model is loaded sucessfully.  My training loss of SFT opensora v1.2 start at about 0.4 ~0.5 . But if ckpt not loaded, the loss starts about 4~5.

**Comment by FrankLeeeee** [comment]
@qian18long do you manage to solve your problem?

**Comment by tinnerhrhe** [comment]
Thanks for your suggestions! The `num_frames` in my own videos ranges from 2 to 7, which is rather smaller compared to the videos used in Open-Sora. After carefully checking the code, I found that the original version only supports videos with more than 17 frames. Therefore, I revised the code related to temporal downsampling. I have attached the loss curve below. Is this curve normal or similar to what you observed during training?
<img width="592" alt="Screenshot 2024-07-10 at 12 43 40 PM" src="https://github.com/hpcaitech/Open-Sora/assets/69794814/9c1938aa-ffb2-4718-9289-ad7be0da9c4e">

Additionally, I have one question about the `timestep_transform` function in [rectified_flow](https://github.com/hpcaitech/Open-Sora/blob/bf4d6673af9407650b2ce6250debc2453b82c572/opensora/schedulers/rf/rectified_flow.py#L10): What are the effects of this function? Similarly, what are the effects of the `scale` arg in [PositionEmbedding2D](https://github.com/hpcaitech/Open-Sora/blob/bf4d6673af9407650b2ce6250debc2453b82c572/opensora/models/layers/blocks.py#L754)? Why you use them? Will they affect the performance if I remove them?

**Comment by CIntellifusion** [comment]
> Thanks for your suggestions! The `num_frames` in my own videos ranges from 2 to 7, which is rather smaller compared to the videos used in Open-Sora. After carefully checking the code, I found that the original version only supports videos with more than 17 frames. Therefore, I revised the code related to temporal downsampling. I have attached the loss curve below. Is this curve normal or similar to what you observed during training? <img alt="Screenshot 2024-07-10 at 12 43 40 PM" width="592" src="https://private-user-images.githubusercontent.com/69794814/347259686-9c1938aa-ffb2-4718-9289-ad7be0da9c4e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA1OTk1MDUsIm5iZiI6MTcyMDU5OTIwNSwicGF0aCI6Ii82OTc5NDgxNC8zNDcyNTk2ODYtOWMxOTM4YWEtZmZiMi00NzE4LTkyODktYWQ3YmUwZGE5YzRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEwVDA4MTMyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2ZTQ2YzFlODA4YmVjMTg1YWVlMTgzZmNjYWE3MzdkOGU1ZDAwYzgxOTdkN2YxY2U4ZDA0YTY0YzI3OTk3OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vhSrN-hq_xiTICM3m06gqTKX_eDggYS_Tqmg7laVtUY">
> 
> Additionally, I have one question about the `timestep_transform` function in [rectified_flow](https://github.com/hpcaitech/Open-Sora/blob/bf4d6673af9407650b2ce6250debc2453b82c572/opensora/schedulers/rf/rectified_flow.py#L10): What are the effects of this function? Similarly, what are the effects of the `scale` arg in [PositionEmbedding2D](https://github.com/hpcaitech/Open-Sora/blob/bf4d6673af9407650b2ce6250debc2453b82c572/opensora/models/layers/blocks.py#L754)? Why you use them? Will they affect the performance if I remove them?

I think this is similar to mine. But I haven't evaluated the my ft ckpt on the vbench yet.

**Comment by Ir1d** [comment]
@tinnerhrhe Were you able to train with videos less than 17 frames? Can you please elaborate how you "revised the code related to temporal downsampling"? Thanks!

**Comment by Space-Xun** [comment]
@qian18long Hi, I also encountered a similar problem, have you solved this problem?
